src/core/initialize.rs

```diff
@@ -80,40 +80,83 @@ impl StackTraceGetter {
     fn get_trace_from_current_thread(&self) -> Result<StackTrace, MemoryCopyError> {
         let stack_trace_function = &self.stack_trace_function;
 
+        let try_lock = |process: &Process| {
+            process
+                 .lock()
+                 .context("locking process during stack trace retrieval")
+                 .or_else(|e| {
+                     if let Some(source) = e.source() {
+                         match source.downcast_ref::<remoteprocess::Error>().ok_or(source) {
+                             Ok(remoteprocess::Error::IOError(e)) => {
+                                 match e.kind() {
+                                     std::io::ErrorKind::NotFound => {
+                                         debug!("Process {} was not found : {:?}", self.process.pid, e);
+                                         return Err(MemoryCopyError::ProcessEnded)
+                                     }
+                                     #[cfg(target_os = "windows")]
+                                     std::io::ErrorKind::PermissionDenied => {
+                                         return Err(MemoryCopyError::ProcessEnded)
+                                     }
+                                     _ => {}
+                                 };
+                             }
+                             #[cfg(target_os = "linux")]
+                             Ok(remoteprocess::Error::NixError(e)) => match e {
+                                 nix::Error::Sys(nix::errno::Errno::EPERM) => {
+                                     debug!("EPERM for process {} : {:?}", self.process.pid, e);
+                                     return Err(MemoryCopyError::ProcessNotLocked);
```

Comment:
I've changed this from ProcessEnded, as I think this is a more appropriate error type - we couldn't lock the process, and got EPERM back. Since this is retriable, we may as well retry, which is the handling for ProcessNotLocked.

Reply:
I ended up having to make a new error type for this, since I think another test depends on treating EPERM as ProcessEnded, which is ultimately treated as Ok downstream, though I think this leads to swallowing an error incorrectly.